server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/services/ActivityService.java

```diff
@@ -0,0 +1,233 @@
+            log.error("Data integrity or constraint violation: {}", e.getMessage(), e);
+            throw e;
+        } catch (EntityNotFoundException e) {
+            log.error("Invalid reference data: {}", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    public ActivityModel getActivity(String projectGuid, String fiscalGuid, String activityGuid) {
+        // Get the activity
+        ActivityEntity activity = activityRepository.findById(UUID.fromString(activityGuid))
+                .orElseThrow(() -> new EntityNotFoundException(ACTIVITY_NOT_FOUND + activityGuid));
+
+        // Verify activity belongs to the specified project fiscal
+        if (!activity.getProjectPlanFiscalGuid().toString().equals(fiscalGuid)) {
+            throw new EntityNotFoundException(ACTIVITY + activityGuid + DOES_NOT_BELONG_FISCAL + fiscalGuid);
+        }
+
+        // Verify project fiscal belongs to project
+        ProjectFiscalEntity projectFiscal = projectFiscalRepository.findById(UUID.fromString(fiscalGuid))
+                .orElseThrow(() -> new EntityNotFoundException(FISCAL_NOT_FOUND + fiscalGuid));
+        if (!projectFiscal.getProject().getProjectGuid().toString().equals(projectGuid)) {
+            throw new EntityNotFoundException(PROJECT_FISCAL + fiscalGuid + DOES_NOT_BELONG_PROJECT + projectGuid);
+        }
+
+        return activityResourceAssembler.toModel(activity);
+    }
+
+    public void deleteActivity(String projectGuid, String fiscalGuid, String activityGuid) {
+        // Get the activity
+        ActivityEntity activity = activityRepository.findById(UUID.fromString(activityGuid))
+                .orElseThrow(() -> new EntityNotFoundException(ACTIVITY_NOT_FOUND + activityGuid));
+
+        // Verify activity belongs to the specified project fiscal
+        if (!activity.getProjectPlanFiscalGuid().toString().equals(fiscalGuid)) {
+            throw new EntityNotFoundException(ACTIVITY + activityGuid + DOES_NOT_BELONG_FISCAL + fiscalGuid);
+        }
+
+        // Verify project fiscal belongs to project
+        ProjectFiscalEntity projectFiscal = projectFiscalRepository.findById(UUID.fromString(fiscalGuid))
+                .orElseThrow(() -> new EntityNotFoundException(FISCAL_NOT_FOUND + fiscalGuid));
+        if (!projectFiscal.getProject().getProjectGuid().toString().equals(projectGuid)) {
+            throw new EntityNotFoundException(PROJECT_FISCAL + fiscalGuid + DOES_NOT_BELONG_PROJECT + projectGuid);
+        }
+
+        activityRepository.deleteById(UUID.fromString(activityGuid));
+    }
+
+    public void assignAssociatedEntities(ActivityModel resource, ActivityEntity entity) {
+        if (resource.getActivityStatusCode() != null) {
+            String forestAreaCode1 = resource.getActivityStatusCode().getActivityStatusCode();
+            ActivityStatusCodeEntity activityStatusCode = loadActivityStatusCode(forestAreaCode1);
```

Comment:
You're only calling the `load<code>` methods in this one spot. Why have them in their own methods? You can just use the code directly here?

Reply:
Changed to access the code directly